src/edu/csus/ecs/pc2/ui/EditProblemPane.java

```diff
@@ -690,232 +700,258 @@ public void enableUpdateButton() {
                             }
                         }
                     }
+                    
+                } else {
+                    logDebugException("No ProblemDataFiles for " + problem);
+                }
+
+
+                // see if the choice of which output validator (if any) to use has changed;
+                // enable the Update button and update the tooltip if so
+                if (problem.getValidatorType() != changedProblem.getValidatorType()) {
+                    enableButton = true;
+                    if (updateToolTip.equals("")) {
+                        updateToolTip = "Output Validator";
+                    } else {
+                        updateToolTip += ", Output Validator";
+                    }
+                }
 
-                    // see if the choice of which output validator (if any) to use has changed;
-                    // enable the Update button and update the tooltip if so
-                    if (problem.getValidatorType() != changedProblem.getValidatorType()) {
+                // see if the PC2 Validator options have changed; enable the Update button and update the tooltip if so
+                if (problem.getPC2ValidatorSettings() != null && changedProblem.getPC2ValidatorSettings() != null) {
```

Comment:
Is this related to a bug fix?   Validator field are not in the scope of sandbox changes.

Reply:
No, there is no change (or intent to change) something related to validators.  The "code change" you are seeing is a result of the "diff" routine being unable to deal with a different change:  altering the scope of this test.  That change was necessary in order to place the code which tests for changes in the sandbox in the correct level of scope.